.github/workflows/docc.yml

```diff
@@ -0,0 +1,46 @@
+name: Deploy Documentation
+on:
+  push:
+    branches: [main, dev]
+  workflow_dispatch:
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: macos-13
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
This could also be 1 as you do not need the whole history

Reply:
What does this do? I haven't looked at it in-depth yet

Reply:
0 would get all the git history, brackens and tags. Default is 1 where it will fetch only the intended branch.